### PR TITLE
Weapons: Show trace for every bullet

### PR DESCRIPTION
**Pull request by EntranceJew**
doesn't make sense to do a dice roll for tracers in a deduction game, why let RNG play that part?

### CODE REVIEW

**Comment by Histalek** [comment]
Sounds good to me, although i'm a bit hesitant given that this has existed for 6 years or rather for nearly as long as TTT2 existed.

Do you have a visual comparison for us? (or at least how it looks after your change?)

**Comment by TimGoll** [comment]
I hardly ever noticed any tracers at all, so I don't care that much. From my POV this can be merged

**Review by TimGoll — APPROVED** [review]

**Review by saibotk — CHANGES_REQUESTED** [review]
This does not change anything with RNG according to the docs. This only affects how many bullet traces are rendered to see the bullets eg. For a shotgun.

See https://wiki.facepunch.com/gmod/Structures/Bullet#Tracer

**Comment by TimGoll** [comment]
> This does not change anything with RNG according to the docs. This only affects how many bullet traces are rendered to see the bullets eg. For a shotgun.
> 
> See https://wiki.facepunch.com/gmod/Structures/Bullet#Tracer

I thin kthat was his point. He doesn't want RNG to effect the tracers

**Comment by saibotk** [comment]
Oh just reread it, this is about drawing every trace. Please adjust the changelog to not mention RNG and explicitly state that it renders every bullet trace.

But to be honest im still against changing this because it looks somewhat stupid for the shotgun maybe adjust the value for that?

Idk maybe we need a visual comparison or just a visual on how a normal weapon shoots vs a shotgun to decide.

**Comment by EntranceJew** [comment]
before, every tracer is a 1/4 chance of drawing per pellet regardless of the weapon:

https://github.com/TTT-2/TTT2/assets/5711436/90c3680c-3a65-4f5e-adf6-8bfa78b0f1b2


https://github.com/TTT-2/TTT2/assets/5711436/acbd427d-7045-4999-9612-a3fa5084d56e


https://github.com/TTT-2/TTT2/assets/5711436/e0604bc6-db68-42be-897d-f8c63a6e1ca9


https://github.com/TTT-2/TTT2/assets/5711436/9fddd188-53d2-493f-893d-2755336f7514

after, every tracer is 1/1 chance of drawing per pellet regardless of the weapon:


https://github.com/TTT-2/TTT2/assets/5711436/13d7f744-2848-4237-9c4e-5fec84685f07


https://github.com/TTT-2/TTT2/assets/5711436/9ca7cb3e-cdb4-474e-9332-14b112a24748


https://github.com/TTT-2/TTT2/assets/5711436/2a40ec18-16fa-40cf-990e-6f2ca5f38be9


https://github.com/TTT-2/TTT2/assets/5711436/2c711363-3d2a-4209-bdbb-399218f7ff4b


https://github.com/TTT-2/TTT2/assets/5711436/fb2039ee-fd39-4bd8-aa62-27b8ae197e60



_note, these occasionally do not draw the tracer BECAUSE they are to close to the surface they impact with and don't pass min clearance for the effect_

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
 - A crosshair is now also drawn when holding a nade, making it less confusing when looking at entities
 - Hides item settings in the equipment editor that are only relevant for weapons
 - The binoculars now use the default crosshair as well
+- Tracers for weapons are no longer RNG 25% chance
```
Comment (Histalek) [inline CHANGELOG.md:27]:
```suggestion
- Tracers are now drawn for every shot/pellet instead of only 25% of shots/pellets
```

**Review by Histalek — CHANGES_REQUESTED** [review]
LGTM besides the changelog nit

**Review by saibotk — APPROVED** [review]
Thank you very much for the videos 👍

**Review by Histalek — APPROVED** [review]

**Pull request merged**